src/ast.ml

```diff
@@ -32,14 +32,14 @@ type uop =
 type expr = 
     NumberLit of float 
   | BoolLit of bool 
-  | StringLit of string 
   | CharLit of char 
+  | StringLit of string 
   | ListLit of expr list 
   | Id of string 
   | Binop of expr * bop * expr
   | Unop of uop * expr
+  | Elem of expr * expr
```

Comment:
might be missing something but what's the significance of `expr * expr` over `string * expr`?

Reply:
We need to figure out list literals, but with the current syntax, `expr * expr` would support something like `[1, 2, 3][0]`